lib/dry/monitor.rb

```diff
@@ -1,20 +1,38 @@
 # frozen_string_literal: true
 
-require "dry/monitor/logger"
-require "dry/monitor/notifications"
 require "dry/core/extensions"
+require "dry/core/constants"
+require "zeitwerk"
+
 
 module Dry
   module Monitor
     extend Dry::Core::Extensions
+    include Dry::Core::Constants
 
     register_extension(:rack) do
       require "rack/utils"
-      require "dry/monitor/rack/logger"
     end
 
     register_extension(:sql) do
-      require "dry/monitor/sql/logger"
+    end
+
+    class << self
+      def root
+        File.expand_path("..", __dir__)
+      end
+
+      def loader
+        @loader||= loader = Zeitwerk::Loader.new.tap do |loader|
```

Comment:
We can't use `Zeitwerk::Loader.for_gem` because of the file nesting, so I setup the loader manually.